code/modules/mob/mob_helpers.dm

```diff
@@ -412,6 +412,7 @@ GLOBAL_LIST_INIT(intents, list(INTENT_HELP,INTENT_DISARM,INTENT_GRAB,INTENT_HARM
 	else
 		if(alert(src, "You sure you want to sleep for a while?", "Sleep", "Yes", "No") == "Yes")
 			SetSleeping(20) //Short nap
```

Comment:
I would rather you stick a `voluntary` arg on this for cleaner code, instead of the assignment below